docs/book/src/reference/creating-events.md

```diff
@@ -0,0 +1,76 @@
+# Creating Events
+
+It is often useful to publish *Event* objects from the controller Reconcile function as they allow users or any automated processes to see what is going on with a particular object and respond to them.
+
+ Recent Events for an object can be viewed by running `$ kubectl describe <resource kind> <resource name>`. Also, they can be checked by running `$ kubectl get events`. 
+
+<aside class="warning">
+<h1>Events should be raised in certain circumstances only</h1>
+
+Be aware that it is **not** recommended to emit Events for all operations. If authors raise too many events, it brings bad UX experiences for those consuming the solutions on the cluster, and they may find it difficult to filter an actionable event from the clutter. For more information, please take a look at the [Kubernetes APIs convention][Events]. 
+
+</aside>
+
+## Writing Events
+
+Anatomy of an Event:
+
+```go
+Event(object runtime.Object, eventtype, reason, message string)
+```
+
+- `object` is the object this event is about.
+- `eventtype` is this event type, and is either *Normal* or *Warning*. [More info][Event-Example]
+- `reason` is the reason this event is generated. It should be short and unique with `UpperCamelCase` format. The value could appear in *switch* statements by automation. [More info][[Reason-Example]]
+- `message` is intended to be consumed by humans. [More info][Message-Example]
+
+
+
+<aside class="note">
+<h1>Example Usage</h1>
+
+Following is an example of a code implementation that raises an Event.
+
+```go
+	// The following implementation will raise an event
+	r.Recorder.Event(cr, "Warning", "Deleting",
+		fmt.Sprintf("Custom Resource %s is being deleted from the namespace %s",
+			cr.Name,
+			cr.Namespace))
+
+```
+
+</aside>
+
+Events are published from a Controller using an [EventRecorder]`type CorrelatorOptions struct`,
```

Comment:
@Sajiyah-Salat now see that you would replace the a, b, c for the steps.
It means here we would have the first step as suggested: https://github.com/kubernetes-sigs/kubebuilder/pull/3375#discussion_r1187046166